python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -535,6 +552,10 @@ def do_evaluate(
                 options.set_delimiter(str(sep))
                 if column_names is not None:
                     options.set_names([str(name) for name in column_names])
+                else:
+                    if not POLARS_VERSION_LT_128:  # pragma: no cover
+                        column_names = read_csv_header(path, str(sep))
+                        options.set_names(column_names)
```

Comment:
What is happening here? Are we somehow losing the header in polars 1.29? Is that due to the change above to `skiprows`?

Reply:
> Is that due to the change above to skiprows?

Yes if you do something like `scan(...).slice(...)`, then skip_rows is set now.

Reply:
I had this wrong before we should only need to read the header here if `skip_rows > header`. Fixed in https://github.com/rapidsai/cudf/pull/18581/commits/273ea91d1809922480870081d5c190027dce1909

Reply:
Can you open a feature req for libcudf to support `skip_rows_after_header` or something like that (which is what we want here, IIUC)?

Reply:
Yup that's exactly what we want, thanks.